### ISSUE TITLE
Support for Web.TV LIVE

### ISSUE THREAD

**Issue by Rotarum** [streamlink/streamlink#142]
Hi,

i have just switched from livestreamer to streamlink. Looks fine so far, but there is a problem with web.tv Site.

http://euronews.web.tv/

The URL for the Euronews Live Broadcast is the following:

https://live2.cdn.web.tv/streams/9e7bdef0a0db8daec2035f0ac64ce239/index.m3u8?st=gtkz6av2_WjhdWPf5Ke1-Q&e=1478645685

Tried streamlink like this:

**streamlink --http-header "Referer=http://euronews.web.tv/" --player-external-http --player-external-http-port 10000 "hls://https://live2.cdn.web.tv/streams/9e7bdef0a0db8daec2035f0ac64ce239/index.m3u8?st=gtkz6av2_WjhdWPf5Ke1-Q&e=1478645685" best**

It says UNABLE TO OPEN URL, maybe have to add something with Browser useragent ??

Hope you can help :)

Greetings

**Comment by beardypig** [comment]
Welcome aboard! I have whipped up a plugin to support web.tv for you, you can try it out - it's under the PR #144

**Comment by Rotarum** [comment]
Thank you 👍 

How can i install and try it ?

**Comment by beardypig** [comment]
Really depends on how you installed `streamlink` and which OS you are using. 

You need to get the new [webtv.py](https://raw.githubusercontent.com/beardypig/streamlink/new-webtv/src/streamlink/plugins/webtv.py) file in to your `streamlink/plugins/` directory.

Which OS are you using and how did you install `streamlink`?

**Comment by Rotarum** [comment]
git clone https://github.com/streamlink/streamlink
sudo python setup.py install

i put the webtv.py into "streamlink/src/streamlink/plugins" and then python setup.py install for reinstalling.

Test again the stream but again "unable to open URL"

What did i wrong ?

**Comment by beardypig** [comment]
Ah cool. OK, you can just change to the new branch ... 
```
git remote add beardypig https://github.com/beardypig/streamlink
git fetch beardypig
git checkout -b new-webtv beardypig/new-webtv
```
Then you will be in a copy my new branch `new-webtv`, then you can install `streamlink` again, that should work. 

You'll need to switch back to the master branch after this branch has been merged, but you can do that by doing
```
git checkout master
```

Let me know if that works or doesn't.

**Comment by Rotarum** [comment]
Ok did exactly how you tell me git remote...fetch...checkout and then install the setup.py

After install i check this folder:

"/usr/local/lib/python2.7/dist-packages/streamlink-0.0.2-py2.7.egg/streamlink/plugins"

In the plugins folder i cannot discover any webtv.py is this normal ?

**Comment by beardypig** [comment]
Try using `pip` rather than invoking `setup.py`.

```
pip uninstall streamlink
pip install --upgrade . # from the streamlink checkout directory 
```

**Comment by Rotarum** [comment]
Finally i got it 👍 

Thank you very much ^^

**Comment by beardypig** [comment]
Cool! Now, how does the plugin work? :)

**Comment by beardypig** [comment]
Also, when you want to switch back to the upstream version you will need to uninstall and reinstall `streamlink` - otherwise it won't get updated :)

**Comment by Rotarum** [comment]
Web.tv works perfect now :)

**Comment by beardypig** [comment]
What do you mean lag?  Buffering or delayed stream or?

**Comment by Rotarum** [comment]
Ok rebooted the Laptop with vlc on it, restart Streamlink and no lag anymore.

In general ffmpeg as client is the better choice. This is what i experienced from former Livestreamer.

Again thank you very much 👍

**Comment by Rotarum** [comment]
Can i request one plugin again, there are 3 sites with similar web live content.

http://canlitv.mobi/
http://www.ecanlitvizle.net/
http://www.canlitvlive.co/

Would like to test them with Streamlink...

Lets hope its possible :)

**Comment by beardypig** [comment]
@Rotarum if you create a new issue for each one that will be better, otherwise they'll get lost in here :)

**Comment by beardypig** [comment]
Fixed in #144
